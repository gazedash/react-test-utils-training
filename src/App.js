import React, { Component } from "react";
import logo from "./logo.svg";
import FollowersSuggestion from "./containers/followersSuggestion"
import "./App.css";

class App extends Component {
  state = {
    background: "grey"
  };

  handleClick = () => {
    this.props.onClick();
    this.setState({ background: "red" });
  };
  render() {
    return (
      <div id="ddd" style={{ background: this.state.background }} className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <button onClick={this.handleClick} />
        <div onClick={() => console.log('new function, right?')} />
        <FollowersSuggestion />
      </div>
    );
  }
}

export default App;
