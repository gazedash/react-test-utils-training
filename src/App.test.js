import React from "react";
import ReactDOM from "react-dom";
import ReactTestUtils from "react-dom/test-utils";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6

import { mount, configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

import App from "./App";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});

it("renders expected children", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const result = renderer.getRenderOutput();

  expect(result.type).toBe("div");
  expect(result.props.children[1].props.children[0]).toBe(
    "To get started, edit "
  );
  expect(result.props.children[1]).toEqual(
    <p className="App-intro">
      To get started, edit <code>src/App.js</code> and save to reload.
    </p>
  );
});

it("simulates click and changes background", () => {
  const mockFunction = jest.fn();
  const element = ReactTestUtils.renderIntoDocument(
    <App onClick={mockFunction} />
  );
  const app = ReactDOM.findDOMNode(element);
  const button = app.children[2];
  ReactTestUtils.Simulate.click(button);
  expect(mockFunction).toHaveBeenCalled();
  expect(app.style.background).toBe("red");
});

it("enzyme: simulate click", () => {
  const mockFunction = jest.fn();
  const element = mount(<App onClick={mockFunction} />);
  element.find("button").simulate("click");
  expect(mockFunction).toHaveBeenCalled();
});

it("matches the snapshot", () => {
  const mockFunction = jest.fn();
  const wrapper = shallow(<App onClick={mockFunction} />);
  // Snapshot serializer will call toJson automatically
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.prop("style")).toMatchSnapshot();
});
