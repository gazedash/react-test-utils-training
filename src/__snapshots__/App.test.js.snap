// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
  style={
    Object {
      "background": "grey",
    }
  }
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <button
    onClick={[Function]}
  />
  <div
    onClick={[Function]}
  />
</div>
`;

exports[`matches the snapshot 2`] = `
Object {
  "background": "grey",
}
`;
